packages/runtime/container-runtime/src/summary/summarizerNode/summarizerNodeWithGc.ts

```diff
@@ -268,21 +286,6 @@ export class SummarizerNodeWithGC extends SummarizerNode implements IRootSummari
 		// If GC is disabled, don't set wip used routes.
 		if (!this.gcDisabled) {
 			wipSerializedUsedRoutes = this.wipSerializedUsedRoutes;
-			/**
-			 * The absence of wip used routes indicates that GC was not run on this node. This can happen if:
-			 * 1. A child node was created after GC was already run on the parent. For example, a data store
-			 * is realized (loaded) after GC was run on it creating summarizer nodes for its DDSes. In this
-			 * case, the used routes of the parent should be passed on the child nodes and it should be fine.
-			 * 2. A new node was created but GC was never run on it. This can mean that the GC data generated
-			 * during summarize is complete . We should not continue, log and throw an error. This will help us
-			 * identify these cases and take appropriate action.
-			 */
-			if (wipSerializedUsedRoutes === undefined) {
-				this.throwUnexpectedError({
```

Comment:
Maybe add a comment to the function saying "does not throw" or something?  Precursor to something we may want to try in Summarizer - switching to `Result` return type that gives strong types for both success and failure paths.

Reply:
switched to return a result.